CI: encouter an issue related to pytest and DB

[link to github action to review the bug](https://github.com/sShaAanGg/library/actions/runs/6574732449/job/17860375971)

related pull request https://github.com/fangci221/library/pull/3

*Uncomment `pip install -r requirement.txt` at line 112, .github/workflows/node.js.yml first to solve the problem of missing module.*

There is a new commit pushed after the pull request but the bug is still unresolved.

If anyone would like to help solve it. Fork this repo first. *Then uncomment `pip install -r requirement.txt` at line 112, .github/workflows/node.js.yml to solve the problem of missing module.*

After commit the above change, the issue now originates from the following code.
```
def connect_mysql():
    
        # Load environment variables from .env
        load_dotenv()
    
        # Set connection parameters
        config = {
            'user': os.getenv('DB_USER'),
            'passwd': os.getenv('DB_PASSWD'),
            'host': os.getenv('DB_HOST'),
>           'port': int(os.getenv('DB_PORT')),
            'db': os.getenv('DB_NAME')
        }
E       TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'

source/test/database.py:16: TypeError
```

--- a/source/back-end/.env
+++ b/source/back-end/.env
@@ -1,4 +1,4 @@
-#  Database
+# Database
 DB_HOST=127.0.0.1
 DB_USER=root
 DB_PASS=123456

--- a/source/test/database.py
+++ b/source/test/database.py
@@ -6,12 +6,12 @@ from dotenv import load_dotenv
 def connect_mysql():
 
     # Load environment variables from .env
-    load_dotenv()
+    load_dotenv(dotenv_path="source/back-end/.env")
 
     # Set connection parameters
     config = {
         'user': os.getenv('DB_USER'),
-        'passwd': os.getenv('DB_PASSWD'),
+        'passwd': os.getenv('DB_PASS'),
         'host': os.getenv('DB_HOST'),
         'port': int(os.getenv('DB_PORT')),
         'db': os.getenv('DB_NAME')

